src/coreclr/jit/regMaskTPOps.cpp

```diff
@@ -59,12 +59,21 @@ bool regMaskTP::IsRegNumInMask(regNumber reg, var_types type) const
 //
 // Parameters:
 //  gprRegs  - Register to check
+//  availableIntRegs - RBM_ALLINT passed in at runtime
 //
+#ifdef TARGET_AMD64
+void regMaskTP::AddGprRegs(SingleTypeRegSet gprRegs, regMaskTP availableIntRegs)
+{
+    assert((gprRegs == RBM_NONE) || ((gprRegs & availableIntRegs) != RBM_NONE));
+    low |= gprRegs;
+}
+#else
 void regMaskTP::AddGprRegs(SingleTypeRegSet gprRegs)
```

Comment:
I think we can reuse the existing method with `DEBUG_ARG()` in it? Earlier I thought we would use `availableIntRegs` value, but if not, we can just reuse the original method.

Reply:
that way you will save adding bunch of `#ifdef` in the caller of this method.

Reply:
Done